### PR TITLE
streaming query testcase in go

### PR DESCRIPTION
**Pull request by saurabh408**
Query stream testcase in go

### CODE REVIEW

File: go/vt/vtgate/endtoend/user_test.go [hunk 1]
```diff
@@ -155,4 +156,59 @@ func TestUsers(t *testing.T) {
...
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer conn.Close()
+
+	//Test insert
+	for i := 1; i <= 4; i++ {
+		each := strconv.Itoa(i)
+		nameQ := "test " + each
```
Comment (deepthi) [inline go/vt/vtgate/endtoend/user_test.go]: what's this variable supposed to mean? It seems weird to end with uppercase Q. Can it be just called name? or nameValue?

File: go/vt/vtgate/endtoend/user_test.go [hunk 2]
```diff
@@ -155,4 +156,59 @@ func TestUsers(t *testing.T) {
...
+			t.Errorf("select:\n%v want\n%v", got, want)
+		}
+	}
+
+	//Test stream over scatter
+	query := "select id, name from vt_user order by id"
+	if err := conn.ExecuteStreamFetch(query); err != nil {
+		t.Fatalf("ExecuteStreamFetch(%v) failed: %v", query, err)
+	}
+	qrS := &sqltypes.Result{}
```
Comment (deepthi) [inline go/vt/vtgate/endtoend/user_test.go]: rename to qr (meaning queryResult)

File: go/vt/vtgate/endtoend/user_test.go [hunk 3]
```diff
@@ -155,4 +156,59 @@ func TestUsers(t *testing.T) {
...
+			t.Fatalf("FetchNext(%v) failed: %v", query, err)
+		}
+		if row == nil {
+			// Done.
+			break
+		}
+		qrS.Rows = append(qrS.Rows, row)
+	}
+	conn.CloseResult()
+	if got, want := fmt.Sprintf("%v", qrS.Rows), `[[INT64(1) VARCHAR("test 1")] [INT64(2) VARCHAR("test 2")] [INT64(3) VARCHAR("test 3")] [INT64(4) VARCHAR("test 4")]]`; got != want {
```
Comment (deepthi) [inline go/vt/vtgate/endtoend/user_test.go]: Can you use assertEqual? or something similar?

**Review by deepthi — CHANGES_REQUESTED** [review]

File: go/vt/vtgate/endtoend/user_test.go (hunk 1 above)
Comment (saurabh408) [reply go/vt/vtgate/endtoend/user_test.go]: Done

File: go/vt/vtgate/endtoend/user_test.go (hunk 2 above)
Comment (saurabh408) [reply go/vt/vtgate/endtoend/user_test.go]: renamed it to qr

File: go/vt/vtgate/endtoend/user_test.go (hunk 3 above)
Comment (saurabh408) [reply go/vt/vtgate/endtoend/user_test.go]: modified test cases using assert Equal

**Comment by saurabh408** [comment]
@deepthi Addressed review comments, please verify

**Review by deepthi — APPROVED** [review]

**Pull request merged**